examples/shader/array_texture.rs

```diff
@@ -0,0 +1,194 @@
+) {
+    if loading_texture.is_loaded
+        || asset_server.get_load_state(loading_texture.handle.clone()) != LoadState::Loaded
+    {
+        return;
+    }
+    loading_texture.is_loaded = true;
+    let image = images.get_mut(&loading_texture.handle).unwrap();
+
+    // Create a new array texture asset from the loaded texture.
+    let array_layers = 4;
+    image.reinterpret_stacked_2d_as_array(array_layers);
+
+    // Spawn some cubes using the array texture
+    let mesh_handle = meshes.add(Mesh::from(shape::Cube { size: 1.0 }));
+    let material_handle = materials.add(ArrayTextureMaterial {
+        array_texture: loading_texture.handle.clone(),
+    });
+    for x in -5..=5 {
+        commands.spawn_bundle(MaterialMeshBundle {
+            mesh: mesh_handle.clone(),
+            material: material_handle.clone(),
+            transform: Transform::from_xyz(x as f32 + 0.5, 0.0, 0.0),
+            ..Default::default()
+        });
+    }
+}
+
+#[derive(Debug, Clone, TypeUuid)]
+#[uuid = "9c5a0ddf-1eaf-41b4-9832-ed736fd26af3"]
+struct ArrayTextureMaterial {
+    array_texture: Handle<Image>,
+}
+
+#[derive(Clone)]
+pub struct GpuArrayTextureMaterial {
+    bind_group: BindGroup,
+}
+
+impl RenderAsset for ArrayTextureMaterial {
+    type ExtractedAsset = ArrayTextureMaterial;
+    type PreparedAsset = GpuArrayTextureMaterial;
+    type Param = (
+        SRes<RenderDevice>,
+        SRes<MaterialPipeline<Self>>,
+        SRes<RenderAssets<Image>>,
+    );
+    fn extract_asset(&self) -> Self::ExtractedAsset {
+        self.clone()
+    }
+
+    fn prepare_asset(
+        extracted_asset: Self::ExtractedAsset,
+        (render_device, material_pipeline, gpu_images): &mut SystemParamItem<Self::Param>,
+    ) -> Result<Self::PreparedAsset, PrepareAssetError<Self::ExtractedAsset>> {
+        let (array_texture_texture_view, array_texture_sampler) = if let Some(result) =
+            material_pipeline
+                .mesh_pipeline
+                .get_image_texture(gpu_images, &Some(extracted_asset.array_texture.clone()))
+        {
+            result
+        } else {
+            return Err(PrepareAssetError::RetryNextUpdate(extracted_asset));
+        };
+        let bind_group = render_device.create_bind_group(&BindGroupDescriptor {
+            entries: &[
+                BindGroupEntry {
+                    binding: 0,
+                    resource: BindingResource::TextureView(array_texture_texture_view),
+                },
+                BindGroupEntry {
+                    binding: 1,
+                    resource: BindingResource::Sampler(array_texture_sampler),
+                },
+            ],
+            label: Some("array_texture_material_bind_group"),
+            layout: &material_pipeline.material_layout,
+        });
+
+        Ok(GpuArrayTextureMaterial { bind_group })
+    }
+}
+
+impl Material for ArrayTextureMaterial {
+    // fn vertex_shader(asset_server: &AssetServer) -> Option<Handle<Shader>> {
```

Comment:
We should remove this comment. I don't like including "unnecessary" commented out code in examples.